pkg/api/default.go

```diff
@@ -1,13 +1,31 @@
 package api
 
 // DefaultVMSizeKubeArguments defines default values of kube-arguments based on the VM size
-var DefaultVMSizeKubeArguments = map[VMSize]map[string]string{
-	StandardD2sV3: map[string]string{
-		"kube-reserved":   "cpu=200m,memory=512Mi",
-		"system-reserved": "cpu=200m,memory=512Mi",
+var DefaultVMSizeKubeArguments = map[VMSize]map[AgentPoolProfileRole]map[string]string{
+	StandardD2sV3: map[AgentPoolProfileRole]map[string]string{
+		AgentPoolProfileRoleMaster: {
+			"system-reserved": "cpu=500m,memory=1Gi",
+		},
+		AgentPoolProfileRoleCompute: {
+			"kube-reserved":   "cpu=200m,memory=512Mi",
+			"system-reserved": "cpu=200m,memory=512Mi",
+		},
+		AgentPoolProfileRoleInfra: {
+			"kube-reserved":   "cpu=200m,memory=512Mi",
+			"system-reserved": "cpu=200m,memory=512Mi",
```

Comment:
as discussed this is a little bit of duplication, but I think we soon will deviate from same values on infra nodes and compute (as soon as our nodes will start running 100 pods and infra still on 10). So this leaves us a place to change. 
I already see a follow-up story. 

We need to monitor system-reserved, kube-reserved and overall node memory blueprint in our stacks to see if these values make sense.